### PR TITLE
Move `_databricks_internal` check to shim layer [databricks]

### PR DESCRIPTION
**Pull request by cindyyuanjiang**
We create a new `DeltaLakeUtils` class with a method `isDatabricksDeltaLakeScan` in three different shims (312db, 321db, 311+-nondb), so that we only perform the `_databricks_internal` check when running on Databricks.

The **problem** was raised in issue #6279.
"PR#6257 added a check for _databricks_internal for 22.08. We need to move check to the shim layer so that we only look for this when running on Databricks."

We adopted the **solution** as above.

### CODE REVIEW

**Comment by cindyyuanjiang** [comment]
build

File: sql-plugin/src/main/312db/scala/com/nvidia/spark/rapids/shims/DeltaLakeUtils.scala [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims
+
+import org.apache.spark.sql.execution.FileSourceScanExec
+
+object DeltaLakeUtils {
+  /* Check for _databricks_internal when running on Databricks */
+  def isDeltaLakeScan(f: FileSourceScanExec): Boolean = {
```
Comment (gerashegalov) [inline sql-plugin/src/main/312db/scala/com/nvidia/spark/rapids/shims/DeltaLakeUtils.scala:23]: There is an OSS version of DeltaLake. Should this be named something like isDatabricksDeltaLakeScan?

File: sql-plugin/src/main/311+-nondb/scala/com/nvidia/spark/rapids/shims/DeltaLakeUtils.scala [hunk 2]
```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2021-2022, NVIDIA CORPORATION.
```
Comment (nartal1) [inline sql-plugin/src/main/311+-nondb/scala/com/nvidia/spark/rapids/shims/DeltaLakeUtils.scala:2]: Only current year should be mentioned for new file. 
```suggestion
 * Copyright (c) 2022, NVIDIA CORPORATION.
```

Comment (cindyyuanjiang) [reply sql-plugin/src/main/311+-nondb/scala/com/nvidia/spark/rapids/shims/DeltaLakeUtils.scala]: Got it, thank you!

File: sql-plugin/src/main/312db/scala/com/nvidia/spark/rapids/shims/DeltaLakeUtils.scala (hunk 1 above)
Comment (cindyyuanjiang) [reply sql-plugin/src/main/312db/scala/com/nvidia/spark/rapids/shims/DeltaLakeUtils.scala:23]: I will update that. Thank you!

**Comment by cindyyuanjiang** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Review by nartal1 — APPROVED** [review]

**Review by abellina — APPROVED** [review]

**Pull request merged**